### PR TITLE
Ciac 604 pr number on rn

### PR DESCRIPTION
**Pull request by JudahSchwartz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/Marketplace/marketplace_services.py [hunk 1]
```diff
@@ -2905,10 +2923,41 @@ def _changelog_entry_bc_versions(predecessor_version: Version, rn_version: Versi
...
+            a dict of version -> List[Pull Request Numbers] for each file in the release_notes_dir
+        """
+
+        def file_path(file):
+            return f"Packs/{self.name}/{self.RELEASE_NOTES}/{file}"
+
+        if not os.path.exists(release_notes_dir):
+            return {}
+        return {underscore_file_name_to_dotted_version(file): get_pull_request_numbers_from_file(file_path(file))
+                for file in filter_dir_files_by_extension(release_notes_dir, '.md')}
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/marketplace_services.py:2942]: As much as it is more elegant it is not very readable, let's create a regular for loop something like this:
```suggestion
        versions_dict = {}
        for file in filter_dir_files_by_extension(release_notes_dir, '.md'):
            rn_version =  underscore_file_name_to_dotted_version(file)
            pr_numbers = get_pull_request_numbers_from_file(file_path(file))
            versions_dict[rn_version] = pr_numbers

            return versions_dict
```

File: Tests/Marketplace/marketplace_services.py [hunk 2]
```diff
@@ -2905,10 +2923,41 @@ def _changelog_entry_bc_versions(predecessor_version: Version, rn_version: Versi
...
+    """
+    Uses git and regex to find the pull request numbers for the given file
+    Args:
+        file_path: The file to find PRs for
+
+    Returns:
+        A list of relevant pull request numbers for the given file
+    """
+    log_info: str = git.Git(CONTENT_ROOT_PATH).log(file_path)
+    return [int(i) for i in re.findall(PULL_REQUEST_PATTERN, log_info)]
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/marketplace_services.py:2958]:
```suggestion
    return [int(pr_number) for pr_number in re.findall(PULL_REQUEST_PATTERN, log_info)]
```
Also, why do we need to cast to int? Is there a reason why we need the pr ids as numbers and not strings?

**Review by ShahafBenYakir — CHANGES_REQUESTED** [review]
Great work.
- Why haven't we tested the get_version_to_pr_numbers function? Let's add a test for it.
- For the future when you are running tests/experiments make sure not to do it on partner packs as it alerts them ([see above](https://github.com/demisto/content/pull/19074#issuecomment-1137108656)) that their pack is being modified, let's change the test to another pack.

File: Tests/Marketplace/marketplace_services.py (hunk 2 above)
Comment (JudahSchwartz) [reply Tests/Marketplace/marketplace_services.py]: doesnt really need to be. Ill change it

**Review by Noy-Maimon — APPROVED** [review]
As discussed, please open the issue regarding the commit pattern

File: Tests/Marketplace/Tests/marketplace_services_test.py [hunk 3]
```diff
@@ -698,6 +695,26 @@ def test_is_the_only_rn_in_block(self, mocker, dummy_pack, version, boolean_valu
         mocker.patch("os.listdir", return_value=dir_list)
         assert is_the_only_rn_in_block(release_notes_dir, version, AGGREGATED_CHANGELOG) == boolean_value
 
+    def test_get_version_to_pr_numbers(self, mocker):
+        dir_list = ['1_0_1.md', '1_0_2.md', '1_0_3.md']
+        mocker.patch("os.listdir", return_value=dir_list)
+        mocker.patch("os.path.exists", return_value=True)
+
+        class GitMock:
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/Tests/marketplace_services_test.py:711]: Why aren't we uniting this with the other GitMock class?

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**